src/friendly-iframe-embed.js

```diff
@@ -452,6 +454,144 @@ export class FriendlyIframeEmbed {
       this.visibilityObservable_.fire(this.visible_);
     }
   }
+
+  /**
+   * @return {!HTMLBodyElement}
+   * @visibleForTesting
+   */
+  getBodyElement() {
+    return /** @type {!HTMLBodyElement} */ (
+        (this.iframe.contentDocument || this.iframe.contentWindow.document)
+            .body);
+  }
+
+  /**
+   * @return {!./service/vsync-impl.Vsync}
+   * @visibleForTesting
+   */
+  getVsync() {
+    return vsyncFor(this.win);
```

Comment:
Thanks for sharing the long story :-). 
Stubbing a vsyncService should not be this hard, which means it should be improved.

That will be out of scope of this PR, but any idea how we can easily fake or stub vsyncService?

PR approved, we can continue that discussion in a new thread.